core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -387,9 +387,15 @@ public boolean sync() throws AccessControlException, InvalidPathException {
           syncPathCount + failedSyncPathCount, syncPathCount, failedSyncPathCount, end - start,
           mRootScheme);
     }
+    boolean success = syncPathCount > 0;
+    if (success) {
+      // update the sync path cache for the root of the sync
+      // TODO(gpang): Do we need special handling for failures and thread interrupts?
+      mUfsSyncPathCache.notifySyncedPath(mRootScheme.getPath().getPath(), mDescendantType);
```

Comment:
So now, if two RPCs on the same path come in at the same time, we'll end up doing a lot more work, right?

Reply:
Yes, that is true. Is there a better way to mark a directory as "synced" only after all the descendants are synced?

Reply:
I think this is fine. A trade-off between memory overhead (keeping track of all inodes in the sync path cache) vs doing more work in unfortunate scenarios